### PR TITLE
Only include transactions with low complexity

### PR DESCRIPTION
**Pull request by catena2w**

### CODE REVIEW

File: src/main/scala/org/ergoplatform/local/ErgoMiner.scala [hunk 1]
```diff
@@ -317,13 +318,14 @@ class ErgoMiner(ergoSettings: ErgoSettings,
     //only transactions valid from against the current utxo state we take from the mem pool
     val emissionTxOpt = ErgoMiner.collectEmission(state, minerPk, ergoSettings.chainSettings.emissionRules).map { tx =>
       implicit val verifier: ErgoInterpreter = ErgoInterpreter(state.stateContext.currentParameters)
-      val cost = state.validateWithCost(tx, Some(upcomingContext)).get
+      val cost = state.validateWithCost(tx, Some(upcomingContext), Int.MaxValue).get
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/local/ErgoMiner.scala]: Why not use maxTransactionComplexity?

File: src/main/scala/org/ergoplatform/local/ErgoMiner.scala [hunk 2]
```diff
@@ -474,7 +477,8 @@ object ErgoMiner extends ScorexLogging {
                     current -> invalidTxs
                   }
               }
-            case _ =>
+            case Failure(e) =>
+              log.debug(s"Do not incude transaction ${tx.id} due to ${e.getMessage}")
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/local/ErgoMiner.scala]:
```suggestion
              log.debug(s"Do not include transaction ${tx.id} due to ${e.getMessage}")
```

File: src/test/scala/org/ergoplatform/nodeView/state/UtxoStateSpecification.scala [hunk 3]
```diff
@@ -44,7 +44,7 @@ class UtxoStateSpecification extends ErgoPropertyTest with ErgoTransactionGenera
       val newBoxes = IndexedSeq(newFoundersBox, rewardBox)
       val unsignedTx = new UnsignedErgoTransaction(inputs, IndexedSeq(), newBoxes)
       val tx: ErgoTransaction = ErgoTransaction(defaultProver.sign(unsignedTx, IndexedSeq(foundersBox), emptyDataBoxes, us.stateContext).get)
-      us.validateWithCost(tx, None).get should be <= 100000L
+      us.validateWithCost(tx, None, Int.MaxValue).get should be <= 100000L
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/nodeView/state/UtxoStateSpecification.scala]: Let's introduce a constant for cost limit long with Constants.DefaultComplexityLimit.

File: src/main/scala/org/ergoplatform/local/ErgoMiner.scala (hunk 1 above)
Comment (catena2w) [reply src/main/scala/org/ergoplatform/local/ErgoMiner.scala]: We'd like to collect emission anyway, even if its complexity is too high.

File: src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala [hunk 4]
```diff
@@ -53,6 +55,75 @@ class ErgoMinerSpec extends FlatSpec with ErgoTestHelpers with ValidBlocksGenera
     empty.copy(nodeSettings = nodeSettings, chainSettings = chainSettings)
   }
 
+  it should "not include too complex transactions" in new TestKit(ActorSystem()) {
```
Comment (oskin1) [inline src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala]: Seems this test produces unstable result

Comment (catena2w) [reply src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala]: Fixed in https://github.com/ergoplatform/ergo/pull/829/commits/e490bebd419e3de0287a4a86d8901b43ebdbdbf0

**Pull request merged**